chapters/chap03-sec02.tex

```diff
@@ -18,17 +18,97 @@ \subsection{\mimc{} Encryption}\label{instantiation:mkhash:mimc-encryption}
+
+Since block ciphers are usually defined over the product space of keys and messages, we consider the variables $c$, \rounds{} and \exponent{} as fixed. We thereby consider an instantiation of $\mimc{}$ with signature
 \begin{align*}
-    \mimcPrime{p} &: \FFx{p} \times \FFx{p} \to \FFx{p} \\
-    \mimc{}_{2^n} &: \FFx{2^n} \times \FFx{2^n} \to \FFx{2^n}
+    \mimcPrime{\rCURVE} &: \FFx{\rCURVE} \times \FFx{\rCURVE} \to \FFx{\rCURVE}
 \end{align*}
 
-From now on, we only consider \mimc{} defined over prime fields. In particular, the field $\FFx{\rCURVE}$ with elements written over $\secpar$ bits, over which \zksnark~operates.
-
 \subsubsection{Security parameters and analysis}\label{instantiation:mkhash:mimc-encryption:security}
 
-To ensure that the exponentiation leads to a permutation in $\FFx{\rCURVE}$, $\exponent{}$ of the form $\exponent{} = 2^t-1$ such that $\gcd(\exponent{}, \rCURVE-1) = 1$. To achieve a security of $\secpar$, we require $\rounds = \left\lceil \frac{\log_2 \rCURVE}{\log_2 \exponent{}} \right\rceil. $\footnote{We do not consider exponents of type $2^t+1$ since the polynomial representing $\mimcEnc$ would be sparse and as such, the number of rounds becomes constant (see~\cite[Section 5.3]{albrecht2016mimc}).}
+In the original design proposed in the $\mimc{}$ paper~\cite{albrecht2016mimc}, the round function is represented as a ``shifted'' permutation via a cubic map (i.e.~the round input message is added to the key and round constant - the ``shift'' -, and a map $x^{\exponent{}}$ permutes the element in the underlying field $\FFx{2^n}$, $n \in \NN$, where $\gcd(\exponent{}, 2^n - 1) = 1$). This function (which is a permutation and therefore invertible) acts as a substitution box (S-box) and brings non-linearity to the scheme, as usually required for security.
+
+In other sections of the paper, however, the $\mimc{}$ authors proposed generalizations to the initial design. These allow $\mimc{}$ to be used:
+\begin{itemize}
+    \item over prime fields of odd characteristic (i.e.~$\FFx{p}$, $p$ odd prime),
+    \item with different permutation polynomials (i.e.~using different exponents in the round function)
+\end{itemize}
+
+Understanding the relationships between these various settings is required in order to use $\mimc{}$ to operate over prime fields of odd characteristic with a non-cubic permutation monomial.
+
+Overall, for $\mimc{}$ to be considered secure, it is important that no attack in the literature (that may provide a significant speedup compared to ``exhaustive key search'') can succesfully be mounted by a $\ppt$ adversary. Two main families of attacks exist: statistical attacks and algebraic attacks.
+
+\paragraph{Statistical attacks.}
+
+In the ``Security Analysis'' section~\cite[Section 4.2]{albrecht2016mimc}, the authors explain that since the cubic function is an APN, linear attacks pose no threat to $\mimc{}$.
```

Comment:
What is an APN? That's not explained in the paper

Reply:
Almost Perfect Nonlinear (functions)
(see the references I cite, i.e. `New Families of Almost Perfect Nonlinear Power Mappings` by Tor Helleseth, Chunming Rong, and Daniel Sandberg, and `Differentially Uniform Mappings for Cryptography` by Kaisa Nyberg - for more info. I don't plan to provide all the background on PN/APN here)

Reply:
I added a footnote to define k-uniformity and APN functions